common/test/acceptance/tests/helpers.py

```diff
@@ -263,6 +247,41 @@ def get_modal_alert(browser):
     return browser.switch_to.alert
 
 
+class EventsTestMixin(object):
+    """
+    Helpers and setup for running tests that evaluate events emitted
+    """
+    def setUp(self):
+        super(EventsTestMixin, self).setUp()
+        self.event_collection = MongoClient()["test"]["events"]
+        self.start_time = datetime.now()
+        self.addCleanup(self.cleanup)
+
+    def assert_event_emitted_num_times(self, event_name, event_time, event_user_id, num_times_emitted):
+        """
+        Tests the number of times a particular event was emitted.
+        :param event_name: Expected event name (e.g., "edx.course.enrollment.activated")
+        :param event_time: Latest expected time, after which the event would fire (e.g., the beginning of the test case)
+        :param event_user_id: user_id expected in the event
+        :param num_times_emitted: number of times the event is expected to appear since the event_time
+        """
+        self.assertEqual(
+            self.event_collection.find(
+                {
+                    "name": event_name,
+                    "time": {"$gt": event_time},
+                    "event.user_id": int(event_user_id),
+                }
+            ).count(), num_times_emitted
+        )
+
+    def cleanup(self):
+        """
+        Remove the events from mongo
+        """
+        self.event_collection.drop()
```

Comment:
I prefer to do this at initialization time since it makes it easier to diagnose issues when the tests fail since you can look in the collection and see what was written to it.